openlibrary/core/lending.py

```diff
@@ -140,24 +140,43 @@ def get_loan(identifier, user_key=None):
     borrowed that book.
     """
     d = web.ctx.site.store.get("loan-" + identifier)
-    if d and (user_key is None or d['user'] == user_key):
-        loan = Loan(d)
-        if loan.is_expired():
-            loan.delete()
-            return
+
+    account = None
+    if user_key:
+        from openlibrary.accounts.model import OpenLibraryAccount
+        if user_key.startswith('@'):
+            account = OpenLibraryAccount.get(link=user_key)
+        else:
+            account = OpenLibraryAccount.get(username=user_key.split('/')[-1])
+
+    if d:
+        if (user_key is None or (d['user'] == account.username) or (d['user'] == account.itemname)):
+            loan = Loan(d)
+            if loan.is_expired():
+                loan.delete()
+                return
     try:
-        return _get_ia_loan(identifier, user_key and userkey2userid(user_key))
+        return _get_ia_loan(identifier, account and userkey2userid(account.username))
```

Comment:
Do we know *for sure* that a failure to find the OL identifier will result in an Exception? i.e. If `_get_ia_loan()` somehow changed to return False or something benevolent in the Not Found case, then this `get_loan()` function will simply return and the application will never attempt to fetch using the IA identifier.

Instead of returning right here on a successful `_get_ia_loan()` call, we could capture the result, and then test for the result being e.g. None, and if it is, then proceed with a search by IA identifier.